runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java

```diff
@@ -431,6 +431,18 @@ else if (config.swapHomePortal() != HouseMode.ENTER && option.equals("enter"))
 					break;
 			}
 		}
+		else if (config.swapTeleportToDestination() != ObeliskMode.ACTIVATE && target.equals("obelisk"))
+		{
+			switch (config.swapTeleportToDestination())
+			{
+				case SET_DESTINATION:
```

Comment:
This isn't what I meant... `ObeliskMode` enum has a toString method , so just call `swap(config.swapTeleportToDestination(), option, target, true)` instead of checking which enum it is.

Reply:
Sorry, not sure what you mean by that.